app/models/deploy_service.rb

```diff
@@ -122,4 +123,25 @@ def notify_outbound_webhooks(deploy)
   def send_sse_deploy_update(type, deploy)
     SseRailsEngine.send_event('deploys', type: type, deploy: DeploySerializer.new(deploy, root: nil))
   end
+
+  # TODO: make this work with remote images too ... and images where the local image no longer exists
+  def tag_docker_images(deploy)
+    return unless ENV["DOCKER_FEATURE"]
+    return unless deploy.succeeded?
+    return unless builds = deploy.project.builds.
+      where(git_sha: deploy.job.commit).where.not(docker_image_id: nil).presence
```

Comment:
Might be a good idea to extract this into a method of `deploy`, and then have a test or two on that method.